code/modules/projectiles/gun.dm

```diff
@@ -357,16 +357,16 @@
 		wielded_fully = TRUE
 		return TRUE
 
-/obj/item/gun/proc/is_wielded()
-	return wielded
-
 /// triggered on unwield of two handed item
 /obj/item/gun/proc/on_unwield(obj/item/source, mob/user)
 	wielded = FALSE
 	wielded_fully = FALSE
 	zoom(user, forced_zoom = FALSE)
 	user.remove_movespeed_modifier(/datum/movespeed_modifier/gun)
 
+/obj/item/gun/proc/is_wielded()
+	return wielded
+
```

Comment:
I forget how much guncode snowflakes its wielding but i would also refactor  this as well to use the trait. you can even just put a HAS_TRAIT in the return of that is_wielded if gun uses it alot for now